### PR TITLE
feat(idea/squid) merge voucher and squid indexers

### LINKED ISSUE #1643: feat(squid) merge voucher indexer with squid
### Problem to Solve

There is no reason to have two different services using subsquid (squid and voucher indexer). It is much easier to have them in one service.

### Possible Solution

Move the voucher indexer to the squid indexer. Voucher API should be moved to explorer or create a simple HTTP server to work with the voucher indexer part.

### Notes

_No response_

### PR DESCRIPTION
**Pull request by Zewasik**
Resolves #1643 .

- combine squid and voucher indexers
- fix squid event handler types
- move http server for voucher to explorer
- add voucher jsonrpc handlers to explorer

@osipov-mit

### CODE REVIEW

File: idea/explorer/src/jsonrpc.ts [hunk 1]
```diff
@@ -110,4 +159,16 @@ export class JsonRpcServer extends JsonRpc(JsonRpcBase) {
...
   }
+
+  @JsonRpcMethod('voucher.all')
+  @Cache(15)
+  async voucherAll(params: ParamGetVouchers) {
+    return this._services.get(params.genesis).voucher.getVouchers(params);
+  }
+
+  @JsonRpcMethod('voucher.data')
+  @Cache(15)
```
Comment (osipov-mit) [inline idea/explorer/src/jsonrpc.ts:170]:
```suggestion
  @Cache(300)
```

File: idea/explorer/src/jsonrpc.ts [hunk 2]
```diff
@@ -110,4 +159,16 @@ export class JsonRpcServer extends JsonRpc(JsonRpcBase) {
   async eventData(params: ParamGetEvent) {
     return this._services.get(params.genesis).event.getEvent(params);
   }
+
+  @JsonRpcMethod('voucher.all')
+  @Cache(15)
```
Comment (osipov-mit) [inline idea/explorer/src/jsonrpc.ts:164]:
```suggestion
  @Cache(60)
```

File: idea/explorer/src/jsonrpc.ts [hunk 3]
```diff
@@ -30,6 +33,52 @@ export class JsonRpcServer extends JsonRpc(JsonRpcBase) {
...
+        res.status(400).json({ error: 'Network is not supported' });
+        return;
+      }
+
+      try {
+        const vouchers = await voucherService.getVouchers(req.body);
+        res.json(vouchers);
+      } catch (error) {
+        res.status(500).json({ error: 'Internal server error' });
+      }
```
Comment (osipov-mit) [inline idea/explorer/src/jsonrpc.ts:80]: Don't you want to refactor this using decorators? As it's done with jsonrpc methods

Comment (Zewasik) [reply idea/explorer/src/jsonrpc.ts:80]: I decided that ideally REST should not be in the JSONRPC server at all and is a temporary solution

File: idea/explorer/src/decorators/method.ts [hunk 4]
```diff
@@ -76,5 +85,38 @@ export function JsonRpc<TBase extends Constructor<JsonRpcBase>>(Base: TBase) {
         };
       }
     }
+
+    createRestRouter(): Router {
+      const router = Router();
+
+      for (const { method, path, handler } of restHandlers) {
+        router[method](path, async (req, res) => {
+          const { genesis } = req.query;
```
Comment (osipov-mit) [inline idea/explorer/src/decorators/method.ts:101]: Perhaps it's better to have `genesis` in the body of the request as is done in jsonrpc requests

**Review by osipov-mit — APPROVED** [review]

**Pull request merged**